packages/explain-plan-helper/src/index.ts

```diff
@@ -23,31 +24,33 @@ export class ExplainPlan {
   totalKeysExamined: number;
   totalDocsExamined: number;
   originalExplainData: Stage;
-  executionStats: ExecutionStats;
+  executionStats?: ExecutionStats;
 
   constructor(originalExplainData: Stage) {
     const rawExplainObject = convertExplainCompat(originalExplainData);
     const executionStats = getExecutionStats(rawExplainObject);
-    ExplainPlan.addParentStages(executionStats.executionStages);
+    if (executionStats?.executionStages) {
+      ExplainPlan.addParentStages(executionStats.executionStages);
+    }
     const qpInfo = getPlannerInfo(rawExplainObject);
     const esInfo =
       executionStats?.executionStages?.shards?.[0] ?? executionStats;
     this.executionStats = executionStats;
     this.namespace = qpInfo.namespace;
     this.parsedQuery = qpInfo.parsedQuery;
     this.executionSuccess = esInfo.executionSuccess;
-    this.nReturned = executionStats.nReturned;
-    this.executionTimeMillis = executionStats.executionTimeMillis;
-    this.totalKeysExamined = executionStats.totalKeysExamined;
-    this.totalDocsExamined = executionStats.totalDocsExamined;
+    this.nReturned = executionStats?.nReturned ?? 0;
```

Comment:
Should we default to `undefined` here and then its easy for us to show *na* instead of 0?

Reply:
Hmmm, good point, I'll change this

Reply:
Done in https://github.com/mongodb-js/compass/pull/4395/commits/b891e0305fb5d1e578ea7bd833af58601c72c856